File: ServerProject/src/main/java/com/example/meditime/dto/ClientWithMedicationsDTO.java
// Amy Wickham 121785021
// File: ClientWithMedicationsDTO.java
// Description: DTO for representing a client and their list of medications.

package com.example.meditime.dto;

import lombok.Getter;
import lombok.Setter;

import java.util.List;

/**
 * Data Transfer Object representing a client along with their list of assigned medications.
 *
 * Contains the client's ID, name, and a list of ClientMedicationDTO objects representing
 * the medications associated with the client.
 */

@Getter
@Setter
public class ClientWithMedicationsDTO {
    private Long clientId;
    private String clientName;
    private List<ClientMedicationDTO> medications;
}

File: Mobile/app/src/main/java/com/example/medicationapp/model/ClientWithMedicationsDTO.kt

package com.example.medicationapp.model

/**
 * Data Transfer Object (DTO) representing a client along with their associated medications.
 * Used for passing combined client and medication data between application layers or components.
 * Contains the client's ID, name, and a list of ClientMedication instances.
 */



data class ClientWithMedicationsDTO(
    val clientId: Long,
    val clientName: String,
    val medications: List<ClientMedication>
)
